### PR TITLE
chore(build) Use xcbeautify for TravisCI build pipeline

### LINKED ISSUE #90: Improve swiftlint reporting
When there is a linting issue while running in travis-ci, the output is hidden in the build log file. This makes it hard to identify what went wrong and can make it confusing and frustrating for contributors.

### PR DESCRIPTION
**Pull request by surpher**
`xcpretty` tool does not seem to be working in our favour anymore on TravisCI when running `xcodebuild` commands.

Suggestion is to replace it with `xcbeautify`.

Changes:
- Replace `xcpretty` with `xcbeautify` for `xcodebuild` command,
- Remove `--no-skip-current` flag from the command that builds the dependencies (when `pact-consumer-swift` fails to build, TravisCI.org _hides_ the output in a `.log` instead of printing it in build console for developer to see what has happened),
- Add a `scripts/build.sh` step that tests the framework builds as a Carthage dependency,
- Remove `--strict` from Swiftlint command to allow the bundle to build.

This PR will resolve #90.

### CODE REVIEW

File: scripts/install_deps.sh [hunk 1]
```diff
@@ -1,6 +1,5 @@
 #!/bin/bash
 set -e
 
-gem install xcpretty
```
Comment (andrewspinks) [inline scripts/install_deps.sh:4]: should we install xcbeautify here? I think that is why the build failed...

File: Brewfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-tap 'pact-foundation/pact-ruby-standalone'
+tap 'pact-foundation/pact-ruby-standalone', 'https://github.com/pact-foundation/homebrew-pact-ruby-standalone.git'
```
Comment (andrewspinks) [inline Brewfile]: Why this change? Does this cause it to always install latest from git?

**Review by andrewspinks — COMMENTED** [review]
looks like xcbeautify needs to be installed on the agents.

File: Brewfile (hunk 2 above)
Comment (surpher) [reply Brewfile]: I saw TravisCI get "stuck" waiting for someone to enter a git username. Apparently if it is `https://` it doesn't need to authenticate. 🤷‍♂ 

This specifies where _exactly_ to find the tap.

File: scripts/install_deps.sh (hunk 1 above)
Comment (surpher) [reply scripts/install_deps.sh:4]: Odd. I had it in `Brewfile`, and agent complained it was already installed. 🤷‍♂ 

Screen Shot 2020-04-06 at 11 17 11 pm

I'll add it in anyway. Better for it to complain to re-install instead of failing the build.

**Comment by surpher** [comment]
Apparently TravisCI doesn't like `Brewfile`s on branches
